src/annotation/annotations_app/views/api.py

```diff
@@ -343,6 +356,206 @@ def image_retrieve(image_id: str):
         )
 
 
+@app.route("/api/v0/images/<string:img_id>/annotations", methods=['GET'])
+@login_required
+def image_annotations(img_id):
+    """List of annotations for a given image
+    ---
+    get:
+      responses:
+        200:
+          description: List of all annotations for the given image
+          content:
+            application/json:
+              schema: AnnotationListSchema
+    """
+    with db_session() as db:
+        image_in_db = db.query(Image).filter(
+            Image.id == img_id,
+            Image.collections.any(ImageCollection.user_id == current_user.id),
+        ).first()
+
+        if not image_in_db:
+            abort(404)
+    
+        queryset = db.query(Annotation).filter(
+            Annotation.img_id == img_id
+        )
+        total = queryset.count()
+        results = queryset.order_by("id")
+
+        return AnnotationListSchema().dump(
+            {
+                "total": total,
+                "images": results,
+            }
+        )
+
+# TODO: geometry and metadata types
+# TODO: multipart/form request?
+@app.route("/api/v0/annotations/", methods=['POST'])
+@login_required
+def annotation_post():
+    """Upload annotation to collection (or without one)
+    ---
+    post:
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  img_id:
+                    type: string
+                  geometry: 
+                    type: object
+                  annotation: 
+                    type: string
+                  metadata: 
+                    type: object                  
+        responses:
+            201:
+              description: Details about the created annotation
+              content:
+                application/json:
+                  schema: AnnotationDisplaySchema
+    """
+    logging.info("Uploading annotation for user %s", current_user.id)
+    
+    with db_session() as db:
+        req = request.get_json()
+        img_id = req["img_id"] if "img_id" in req else None
```

Comment:
I think you forgot to raise exception (404/400) if `not img_id` after this line; although next like would probably effectively do it but why borther the database and risk finding some object with empty `id` for some reason

Reply:
re the migrations, yeah I didn't want to mess with your setup but I agree it would be best to just create things cleanly from scratch. We should probably explicitly name the primary key and foreign key constraints so we can adjust them later without too much pain.

Reply:
my setup has no data which can't be recreated quickly using that nice Swagger UI, and doing that I test the fact that it still works so no problems :-)